src/library/PiaFormat.ts

```diff
@@ -0,0 +1,206 @@
+          piaData.assign(fieldFormat.name, gender);
+          break;
+        case PIATypes.piaInt:
+          break;
+        case PIATypes.piaDecimal:
+          break;
+        case PIATypes.piaFloat:
+          break;
+        case PIATypes.piaMonthYear:
+          break;
+        case PIATypes.piaString:
+          piaData.assign(fieldFormat.name, dataStr);
+          break;
+      }
+    });
+  }
+  return piaData;
+}
+
+// const constructPieceType = (value: string, ourType: PIATypesEnum) => {
+//   switch (ourType) {
+//     case PIATypesEnum.piaDate:
+//       if (value.length !== 8) {
+//         console.warn("piaFormat date is invalid length");
+//       }
+//       /* dayjs will use local timezone */
+//       /* starts in mmddyyyy, dayjs takes yyyy-mm-dd */
+//       return dayjs(
+//         [value.slice(4, 8), value.slice(0, 2), value.slice(2, 4)].join("-")
+//       ).toDate();
+//       break;
+//     case PIATypesEnum.piaInt:
+//       return new Number(value);
+//       break;
+//     case PIATypesEnum.piaString:
+//     default:
+//       return value;
+//       break;
+//   }
+//   return;
+// };
+
+export class PiaFormat {
+  piaAll: string;
+  piaData: PIAData;
+  constructor(piaInput: string, fileName: string) {
+    this.piaAll = piaInput;
+    const lines = piaInput.split("\n");
+    this.piaData = new PIAData();
+    lines.forEach((line) => {
+      this.piaData = parsePiaLine(line, this.piaData);
+    });
+  }
+
+  outputPIA() {
+    return [
+      "01",
+      this.piaData.assign,
+      this.piaData.sex,
+      dayjs(this.piaData.birthDate).format("MMDDYYYY"),
+      "\n",
+      //this.piaEverythingElse.join("\n"),
+    ].join("");
+  }
+}
+
+// export class PiaFormat {
+//   constructor(piaInput: string, fileName: string) {
+//     this.piaAll = piaInput;
+//     const lines = piaInput
+//       .split("\n")
+//       .map((n) => [n.slice(0, 2), n.slice(2)])
+//       .map((m) => piaLineParser(m[0], m[1]));
+
+//     /* we should not index this array from zero
+//     because the file format does not, just roll them
+//     into one object for every '01', '03', etc. line
+//     if it exists */
+//     this.ssn = lines[0][1][0].value;
+//     this.sex = lines[0][1][1].value;
+//     this.birthDate = lines[0][1][2].value;
+//     this.piaEverythingElse = lines.slice(1).map((n) => n.join(""));
+
+//     console.log(fileName, lines, this.birthDate, this.piaEverythingElse);
+//   }
+//   piaAll: string | undefined;
+//   piaEverythingElse: Array<string>;
+//   ssn: string;
+//   sex: Number;
+//   birthDate: Date;
+
+// }
+
+function parsePiaMonth(ymStr: string): Date {
+  var djs = dayjs();
+  let year = ymStr.slice(0, 2);
+  let month = ymStr.slice(2);
+  djs
+    .set("month", parseInt(month, 10))
+    .set("year", parseInt(year, 10))
+    .set("day", 1);
+  return djs.toDate();
+}
+
+function parsePiaGender(genderStr: string): PIAGender {
+  let genInt = parseInt(genderStr);
+  let gender: PIAGender = genInt;
```

Comment:
We're in danger here if `genInt` is not 0 or 1. We should probably throw an error here if that's the case; the easiest way to check is to look up the value in the enum:
```
function parsePiaSex(str: string): PIASex {
  const sexInt = parseInt(str)
  if (PIASex[sexInt] === undefined) throw Error(`Invalid value for sex supplied: ${sexInt}`)
  return sexInt
}